src/cli/self_update.rs

```diff
@@ -680,7 +680,11 @@ fn install_bins() -> Result<()> {
     if rustup_path.exists() {
         utils::remove_file("rustup-bin", &rustup_path)?;
     }
-    utils::copy_file(&this_exe_path, &rustup_path)?;
+    if cfg!(feature = "no-self-update") {
+        utils::symlink_file(&this_exe_path, &rustup_path)?;
```

Comment:
I'm tempted to request that you emit a warning here that we are making a symbolic link not a copy, and as such, if the location of `rustup-init` changes in future, the installation may cease to work.

Reply:
I'm okay with a warning. I've added a second commit that put's a warning in the pre-install message. Let me know what you think of it.